.eleventy.js

```diff
@@ -185,111 +178,65 @@ module.exports = function (eleventyConfig) {
+        const page = await browser.newPage();
+
+        // Read all files in the input directory
+        const files = fs.readdirSync(inputDir);
+
+        for (const filename of files) {
+          if (filename.endsWith(".svg")) {
+            const inputPath = path.join(inputDir, filename);
+            const outputPath = path.join(outputDir, filename.replace('.svg', '.jpg'));
+
+            // Read the SVG content
+            const svgContent = fs.readFileSync(inputPath, 'utf8');
+
+            // Extract width and height from SVG (Optional: If SVG has explicit size)
+            const matchWidth = svgContent.match(/width="([0-9]+)"/);
+            const matchHeight = svgContent.match(/height="([0-9]+)"/);
+
+            const width = matchWidth ? parseInt(matchWidth[1], 10) : 1200; // Default to 1200px
+            const height = matchHeight ? parseInt(matchHeight[1], 10) : 675; // Default to 630px
+
+            // Set the viewport size to match SVG size
+            await page.setViewport({ width, height });
+
+            // Set SVG content inside an HTML wrapper
+            await page.setContent(`
+                      <html>
+                          <body style="margin:0;padding:0;overflow:hidden;">
+                              <div style="width:${width}px; height:${height}px;">
+                                  ${svgContent}
+                              </div>
+                          </body>
+                      </html>
+                  `);
+
+            // Take a screenshot and save as JPEG
+            await page.screenshot({
+              path: outputPath,
+              type: 'jpeg',
+              quality: 100,
+              clip: { x: 0, y: 0, width, height } // Ensure clipping matches viewport
+            });
+
+            console.log(`Converted: ${filename} -> ${outputPath}`);
+          }
         }
-      }
 
-      await browser.close();
-    }
-
-    // Execute conversion
-    const inputDir = 'dist/assets/images/social-preview-images/';
-    const outputDir = 'dist/assets/images/social-preview-images/';
-    await convertSvgToJpeg(inputDir, outputDir);
-  });
-
-  // Allows you to debug a json object in eleventy templates data | stringify
-  eleventyConfig.addFilter("stringify", (data) => {
-    return JSON.stringify(data, null, "\t");
-  });
+        await browser.close();
+      }
 
-    eleventyConfig.addFilter("customSlug", function (value) {
-        if (!value) return "fallback-title"; // Fallback for empty titles
-        return slugify(value, {
-            lower: true,                 // Convert to lowercase
-            remove: /[^\w\s-]/g          // Remove all non-word characters except spaces and dashes
-        }).replace(/\s+/g, '-');       // Replace spaces with dashes (extra safety)
+      // Execute conversion
+      const inputDir = 'dist/assets/images/social-preview-images/';
+      const outputDir = 'dist/assets/images/social-preview-images/';
```

Comment:
This is using `fs`, so this needs to use `path.join('...', '...', '...');`.

You may want to update the `path` and `fs` import to `node:` to prevent any package from providing those features down the line (security).

Reply:
Heb deze meegenomen in #556 ! Deze PR is alweer even geleden 😬